Added package for PDCurses

jjcamp opened the pull request:
Per #188, adds PDCurses package.

Since the preferred way to build PDCurses on Windows is to use the included NMake file, this implementation simply calls NMake and then copies the relevant files.

Commit:
Added package for PDCurses

msftclas commented:
Hi __@jjcamp__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, MSBOT;

ras0219-msft commented on ports/pdcurses/portfile.cmake:
```diff
@@ -0,0 +1,26 @@
+find_program(NMAKE nmake)
+
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://sourceforge.net/projects/pdcurses/files/latest/download?source=files"
```
When pdcurs releases a new version, this link will conflict with the below hash -- could you switch this out for a stable URL?

Commit:
Changed the download link to a static link.
The different download link packaged the zip slightly differently,
changing the hash as well as the src path.

jjcamp commented:
I can't believe I missed that.  It is fixed now.

ras0219-msft commented:
Thanks for fixing that. After building locally, I noticed a few more issues:

- We require debug builds to be built and available under `debug\lib\` and `debug\bin\`. 
- We prefer DLLs by default if they are available. Static libs should be built when `VCPKG_LIBRARY_LINKAGE` is set to `"static"`.

jjcamp committed:
Re-wrote portfile to support dynamic libraries

jjcamp committed:
Removed panel library.
Panel is not emitted as a dll, and is actually just a copy of the pdcurses library anyway.

jjcamp committed:
Added a fatal error for 64-bit builds.

jjcamp commented:
DLLs and debug builds are now supported.

However, the linter complains about the static library's ar header on my machine, which appears to be 68 bytes instead of the standard 60.  I checked another machine where I had been using the static library for some time, and the header was the same (both are 64-bit Windows 10).

I'm not sure if this is a compiler bug or if vcpkg should be allowing an extra 8 bytes.

Orvid commented on ports/pdcurses/portfile.cmake:
```diff
@@ -0,0 +1,59 @@
+include(${CMAKE_TRIPLET_FILE})
+
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL x64)
```
It might be nice if we could express this in the control file.

ras0219-msft replied:
Opened an issue to track this: https://github.com/Microsoft/vcpkg/issues/229

jjcamp commented:
Update to the linter issue, I just had some bad build artifact cached.  I haven't seen the issue with several fresh builds.

ras0219-msft commented:
Thanks for the fixes!

I added in lines to deploy the `.lib` files during `DLL` mode (they're required on Windows to link).
